platform/api/azure/image.go

```diff
@@ -72,6 +75,84 @@ func IsConflictError(err error) bool {
 	return ok && azerr.Code == "ConflictError"
 }
 
+//go:embed gallery-image-template.json
+var galleryImageTemplate []byte
+
+const (
+	deploymentName    = "kolagalleryimage"
+	galleryNamePrefix = "kolaSIG"
+	imageVersion      = "0.0.0"
+)
+
+type paramValue struct {
+	Value string `json:"value"`
+}
+type galleryParams struct {
+	GalleriesName       paramValue `json:"galleries_name"`
+	ImageName           paramValue `json:"image_name"`
+	ImageVersion        paramValue `json:"image_version"`
+	StorageAccountsName paramValue `json:"storageAccounts_name"`
+	VhdUri              paramValue `json:"vhd_uri"`
+	Location            paramValue `json:"location"`
+	Architecture        paramValue `json:"architecture"`
+	HyperVGeneration    paramValue `json:"hyperVGeneration"`
+}
+
+func azureArchForBoard(board string) string {
+	switch board {
+	case "amd64-usr":
+		return "x64"
+	case "arm64-usr":
+		return "Arm64"
+	}
+	return ""
+}
+
```

Comment:
Maybe we could just add a docstring to explain the difference between `CreateGalleryImage` and `CreateImage` ?
```suggestion
// CreateGalleryImage creates ...
```

Reply:
I added docstrings to the two functions, but there's not much to describe here: these are two different constructs that create an *image* that can be used to spawn VMs. There are a ton of differences between them, but they are irrelevant to this use case.